ui/timeline/build.gradle.kts

```diff
@@ -39,10 +39,13 @@ kotlin {
         val commonMain by getting {
             dependencies {
                 implementation(projects.domain.timeline)
+                implementation(projects.data.persistence)
+                implementation(projects.data.repository)
```

Comment:
I don't feel strongly, but we _should_ be consistent. The ui/* modules currently only should depend on domain/*, and not `data/*` directly. 

This does mean there needs to be a (sometimes redundant) UseCase/wrapper around the repository methods. But I like to think of it as: "If this view/screen/flow only needs access to a single method on my repository, should it really have access to all available APIs on the repo?"

Alternatively, if we move to a full end-to-end feature module architecture, then I am guessing the arch will look like:

app-android -> feature/timeline
feature/timeline -> domain/timeline, data/timeline, util/timeline, .... etc

Reply:
sure I can make the middle domain layer for consistency